### PR TITLE
Implement --quiet / -q

### PR DESCRIPTION
**Pull request by pde**
Which silences all non-error output.

### CODE REVIEW

**Comment by pde** [comment]
Closes #2512.

**Comment by pde** [comment]
@bmw this should be ready for review.

**Comment by bmw** [comment]
@pde, travis is red.

**Comment by pde** [comment]
@bmw it should have been green, I think I had just confused Travis.

File: letsencrypt/main.py [hunk 1]
```diff
@@ -672,7 +672,10 @@ def main(cli_args=sys.argv[1:]):
     sys.excepthook = functools.partial(_handle_exception, config=config)
 
     # Displayer
-    if config.noninteractive_mode:
+    if config.quiet:
+        config.noninteractive_mode = True
+        displayer = display_util.NoninteractiveDisplay(open("/dev/null", "w"))
```
Comment (bmw) [inline letsencrypt/main.py]: nit: We should probably use `os.devnull` here.

File: letsencrypt/reporter.py [hunk 2]
```diff
@@ -76,9 +77,10 @@ def print_messages(self):
         if not self.messages.empty():
             no_exception = sys.exc_info()[0] is None
             bold_on = sys.stdout.isatty()
-            if bold_on:
-                print(le_util.ANSI_SGR_BOLD)
-            print('IMPORTANT NOTES:')
+            if not self.config.quiet:
```
Comment (bmw) [inline letsencrypt/reporter.py]: This logic isn't quite right.

Before, `bold_on` was set and `le_util.ANSI_SGR_BOLD` was written to the terminal depending on the value of `bold_on`. Now, `bold_on` can be set to `True` but we never actually write `le_util.ANSI_SGR_BOLD` to the terminal, causing us to unnecessarily write `le_util.ANSI_SGR_RESET` later.

This isn't a big deal, but it should probably be written correctly.

**Comment by bmw** [comment]
My biggest comment on this PR is I think using `notification` for `_renew_describe_results` looks pretty bad. Here's example output:
```
# letsencrypt renew --force-renewal

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/example.com.conf
-------------------------------------------------------------------------------

-------------------------------------------------------------------------------
new certificate deployed with reload of apache server; fullchain is
/etc/letsencrypt/live/example.com/fullchain.pem
-------------------------------------------------------------------------------

-------------------------------------------------------------------------------

Congratulations, all renewals succeeded. The following certs have been renewed:
  /etc/letsencrypt/live/example.com/fullchain.pem (success)
-------------------------------------------------------------------------------
#
```
If you're happy with this, I suppose we can ignore this problem and move forward with this PR.

**Comment by pde** [comment]
OK, I've tweaked things further so that the `.notification()` call only happens, typical output now looks like:
```
lel renew  --dry-run 

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/isnot.org.conf
-------------------------------------------------------------------------------
2016-04-02 19:29:48,156:WARNING:letsencrypt.renewal:Attempting to renew cert from /etc/letsencrypt/renewal/isnot.org.conf produced an unexpected error: 'server'. Skipping.

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/goxogle.com.conf
-------------------------------------------------------------------------------

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/demo.reworld.org.conf
-------------------------------------------------------------------------------
2016-04-02 19:29:57,768:WARNING:letsencrypt.renewal:Attempting to renew cert from /etc/letsencrypt/renewal/demo.reworld.org.conf produced an unexpected error: 'server'. Skipping.

-------------------------------------------------------------------------------
Processing /etc/letsencrypt/renewal/isnot.org-0001.conf
-------------------------------------------------------------------------------
** DRY RUN: simulating 'letsencrypt renew' close to cert expiry
**          (The test certificates below have not been saved.)

The following certs were successfully renewed:
  /etc/letsencrypt/live/goxogle.com/fullchain.pem (success)
  /etc/letsencrypt/live/isnot.org-0001/fullchain.pem (success)

The following certs could not be renewed:
  /etc/letsencrypt/live/isnot.org/fullchain.pem (failure)
  /etc/letsencrypt/live/demo.reworld.org/fullchain.pem (failure)
** DRY RUN: simulating 'letsencrypt renew' close to cert expiry
**          (The test certificates above have not been saved.)
2 renew failure(s), 0 parse failure(s)
```

I think having the "heading" around the start of each renewed lineage's logs is probably nicer than what we have in `master` at the moment.

**Comment by bmw** [comment]
@jsha: [Error creating new cert](https://api.travis-ci.org/jobs/120326815/log.txt)

File: letsencrypt/reporter.py (hunk 2 above)
Comment (bmw) [inline letsencrypt/reporter.py]: This doesn't seem to have been addressed.

File: letsencrypt/reporter.py [hunk 3]
```diff
@@ -86,6 +88,11 @@ def print_messages(self):
...
         while not self.messages.empty():
             msg = self.messages.get()
+            if self.config.quiet:
+                # In --quiet mode, we only print high priority messages that
+                # are flagged for crash cases
+                if not (msg.priority == self.HIGH_PRIORITY and msg.on_crash):
+                    continue
             if no_exception or msg.on_crash:
                 if bold_on and msg.priority > self.HIGH_PRIORITY:
                     sys.stdout.write(le_util.ANSI_SGR_RESET)
```
Comment (bmw) [inline letsencrypt/reporter.py]: That's still not right due to this line.

Comment (pde) [inline letsencrypt/reporter.py]: Thanks.  I was reasoning incorrectly about that state machine!

**Comment by bmw** [comment]
Haha. I think a better way to have approached the problem is only set `bold_on` to `True` if we print `le_util.ANSI_SGR_BOLD`, but this works.

That is assuming you don't want to use bold text for high priority messages when using `--quiet`. I don't really see a reason for not using bold text if it's available (stdout is a tty), but I don't have a problem if this is the behavior you want.

If this is the desired behavior, this LGTM if it looks good to Travis.

**Comment by pde** [comment]
I don't think it matters very much; people running with `--quiet` are presumably not looking for nicely formatted output :)

**Pull request merged**